### PR TITLE
Repair TYPE_CHECKING blocks

### LINKED ISSUE #31: crashes on TYPE_CHECKING block that has no imports
```py
import typing


if typing.TYPE_CHECKING:
    print("hi")


```

```
Traceback (most recent call last):
  File "/home/classic/.venv3/bin/zimports", line 33, in <module>
    sys.exit(load_entry_point('zimports', 'console_scripts', 'zimports')())
  File "/home/classic/dev/zimports/zimports/cli.py", line 110, in main
    run_with_options(options)
  File "/home/classic/dev/zimports/zimports/zimports.py", line 854, in run_with_options
    _run_file(options, filename)
  File "/home/classic/dev/zimports/zimports/zimports.py", line 798, in _run_file
    result, stats = Rewriter(options, filename, source_lines).rewrite()
  File "/home/classic/dev/zimports/zimports/zimports.py", line 160, in rewrite
    rewritten = self._do_rewrite(self.source_lines, type_check_pass=True)
  File "/home/classic/dev/zimports/zimports/zimports.py", line 103, in _do_rewrite
    imports, warnings, lines_with_code = _parse_toplevel_imports(
  File "/home/classic/dev/zimports/zimports/zimports.py", line 459, in _parse_toplevel_imports
    tree = ast.parse(source, filename)
  File "/opt/python-3.10.0/lib/python3.10/ast.py", line 50, in parse
    return compile(source, filename, mode, flags,
  File "test3.py", line 5
    print("hi")
IndentationError: unexpected indent

```

### LINKED ISSUE #33: crashes on TYPE CHECKING with ELSE
e.g.

```py
if TYPE_CHECKING:
   # etc.
else:
   # etc.
```

### PR DESCRIPTION
**Pull request by zzzeek**
this is a little nutty, I was going to try to just skip blocks that included elses...but lets see how this goes.

Fixes: #31 
Fixes: #33

### CODE REVIEW

**Comment by zzzeek** [comment]
this likely should also add a command line option to disable it trying to affect TYPE_CHECKING blocks if there's more problems.

**Comment by CaselIT** [comment]
Moar passes :D

will take a look tomorrow

File: zimports/zimports.py [hunk 1]
```diff
@@ -26,6 +27,12 @@
 from .vendored.flake8 import normalize_path
 
 
+class RewritePass(enum.Enum):
+    PLAIN = 0
+    TYPE_CHECK = 1
+    ANTI_TYPE_CHECK = 2
```
Comment (CaselIT) [inline zimports/zimports.py:33]: just fyi, you can import `auto` from the enum module and use it to create the values here
```py
class RewritePass(enum.Enum):
    PLAIN = auto()
    TYPE_CHECK = auto()
    ANTI_TYPE_CHECK = auto()
```

File: zimports/zimports.py [hunk 2]
```diff
@@ -172,29 +197,79 @@ def rewrite(self):
...
-                    # line is not empty and not an indented import so the
-                    # type checking import block has ended
-                    in_type_checking_block = False
+                if line and re.match(r"^(?:else|elif)", line):
+                    # it's an else.  new typing block
+                    in_anti_type_checking_block = True
+                    self.anti_type_checking_blocks.append(
+                        (lineno, set(), self.type_checking_blocks[-1][0])
+                    )
+                elif line and re.match(r"^[a-zA-Z]", line):
```
Comment (CaselIT) [inline zimports/zimports.py:220]: number is also legal here. not very useful but technically not a syntax error:

```py
if TYPE_CHECKING:
   ...
123
```

**Review by CaselIT — COMMENTED** [review]
looks ok. now we need to came up with an anti-plain pass :)

File: zimports/zimports.py (hunk 1 above)
Comment (zzzeek) [reply zimports/zimports.py:33]: yeah I like the integers better at the moment

File: zimports/zimports.py (hunk 2 above)
Comment (zzzeek) [reply zimports/zimports.py:220]: really?  hm .   well tool works for us :)

File: zimports/zimports.py (hunk 1 above)
Comment (CaselIT) [reply zimports/zimports.py:33]: btw auto would returns numbers.

File: zimports/zimports.py (hunk 2 above)
Comment (zzzeek) [reply zimports/zimports.py:220]: bzzt.  can also be an underscore.   see #34

Comment (CaselIT) [reply zimports/zimports.py:220]: I guess it could also start with `-` in a negative number. But that's hardly common

Comment (zzzeek) [reply zimports/zimports.py:220]: well that's just crazy talk.  Im sure **that** will never happen!

Comment (CaselIT) [reply zimports/zimports.py:220]: also `~` is a starting legal char!

**Pull request merged**